xarray/core/indexing.py

```diff
@@ -138,18 +138,26 @@ def group_indexers_by_index(
     options: Mapping[str, Any],
 ) -> list[tuple[Index, dict[Any, Any]]]:
     """Returns a list of unique indexes and their corresponding indexers."""
+    # import here instead of at top to guard against circular imports
+    from xarray.core.indexes import PandasIndex
+
     unique_indexes = {}
     grouped_indexers: Mapping[int | None, dict] = defaultdict(dict)
 
     for key, label in indexers.items():
         index: Index = obj.xindexes.get(key, None)
+        if (key in obj.coords) and (index is None):
+            # TODO: should we raise a more informative error
```

Comment:
We could explicitly check for the 1-d case here before index creation and keep using the conditional raise below for the other cases.